scripts/api-workflow.py

```diff
@@ -0,0 +1,188 @@
+'''script to run during the devfile/api workflows'''
+import argparse
+import re
+import os
+import typing
+import shutil
+
+def get_version_breakdown(version: str, version_type: str) -> typing.Tuple[int, int, int]:
+  '''get the version breakdown'''
+  major, minor, bug_fix = re.search(r'^(\d+)\.(\d+)\.(\d+)', version).groups()
+
+  if not (isinstance(major, str) and isinstance(minor, str) and isinstance(bug_fix, str)):
+    raise TypeError(f'{version_type} could not be read: {version}')
+
+  return int(major), int(minor), int(bug_fix)
+
+def update_stable_versions(release: bool, version: str) -> typing.List[str]:
+  '''update stable-versions.txt'''
+  # Parse the schema version
+  major, minor, bug_fix = get_version_breakdown(version, 'Inputted version')
+
+  cwd = os.getcwd()
+  stable_versions: typing.List[str] = []
+
+  with open(f'{cwd}/libs/docs/src/config/stable-versions.txt', encoding='UTF-8') as f:
+    for line in f.readlines():
+      line = line.strip()
+      line_major, line_minor, line_bug_fix = get_version_breakdown(line, 'stable-version')
+
+      # Cannot change a stable version unless its a new release
+      if release and (line_major is major) and (line_minor is minor):
+        # Cannot change a bug fix version unless its the same bug fix version or newer
+        if line_bug_fix <= bug_fix:
+          is_bug_fix = True
+          stable_versions.append(version)
+        else:
+          raise RuntimeError(f'Version {version} needs to be the same version or newer than {line}')
+
+      else:
+        stable_versions.append(line)
+
+  if release:
+    # New stable version
+    if not is_bug_fix:
```

Comment:
I think this might cause problems if we ever need to publish bugfix versions for older schemas (e.g. 2.1.x). 

Pretend the current stable version is 2.2.0, and we need to publish a bug fix for 2.1.x (meaning 2.1.1 would be published). `is_bug_fix` would be set to false (since the minor version `1` is not equal to `2`). But since we're publishing a release, `2.1.1` would get appended to the stable version list, even though it's not the current stable version (2.2.0 in this case)